protostar/formatter/formatting_result.py

```diff
@@ -1,82 +1,72 @@
-from typing import List
+from abc import ABC
 from dataclasses import dataclass
 from pathlib import Path
 
-from protostar.io.log_color_provider import log_color_provider
+from protostar.io import Message, LogColorProvider
 
 
 @dataclass
-class FormattingResult:
+class FormattingResult(Message, ABC):
     filepath: Path
+    check: bool
+
+    def _format_path(self, fmt: LogColorProvider):
+        return fmt.colorize("GRAY", str(self.filepath))
 
 
 @dataclass
 class BrokenFormattingResult(FormattingResult):
     exception: Exception
 
+    def format_human(self, fmt: LogColorProvider) -> str:
+        header = _pad_header(_get_broken_header(fmt), self.check, fmt)
+        return f"{header} {self._format_path(fmt)}\n{str(self.exception)}"
```

Comment:
I'd probably create separate classes in `protostar.commands`. It's a tradeoff between [low coupling](https://en.wikipedia.org/wiki/Coupling_(computer_programming)) and boilerplate. In layered architectures, application code should be decoupled from I/O.

#### protostar.commands
```py
class BrokenFormattingResultMessage(Message[BrokenFormattingResult]):
    def format_human(self, obj: BrokenFormattingResult) -> str:
        return "..."

    def format_object(self, obj: BrokenFormattingResult) -> dict
        return {}
```

Reply:
discussed this privately